data_structures/queue/queue.c

```diff
@@ -0,0 +1,273 @@
+/**
+ * \file: queue.c
+ * \brief Data structure implementation of queue
+ * \author [shubham patil](https://github.com/shubhamdp)
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+/* Content of header file begins */
+
+/** Structure of a node in queue */
+struct node
+{
+    /** Pointer data in a node */
+    void *data;
+    /** Pointer to next node in queue */
+    struct node *next;
+};
+
+/** Structure of a queue */
+struct Queue
+{
+    /** Length of a queue */
+    int length;
+    /** Head pointer of a queue */
+    struct node *head;
+    /** Tail pointer of a queue */
+    struct node *tail;
+};
+
+/**
+ * Prototype for Queue Data Free Function
+ *
+ * \param[in] data arguments to free function, this will be the data entry in
+ * queue
+ */
+typedef void (*QueueDataFreeFunc)(void *data);
+
+/**
+ * Creates and initialise the queue
+ *
+ * \return pointer pointer to newly created queue
+ * \return NULL in case of no memory
+ */
+extern struct Queue *queue_new();
```

Comment:
`extern` keyword informs the compiler to look for the function in some external file or library. Which one? This is being used throughout the file and it would not compile.

NB: This repo is meant for exceptional purposes i.e., implementations for readers to learn from and not using available libraries.

Reply:
In that case I think, we already have one implementation of queue. Closing this PR.